crates/pathfinder/src/state/sync.rs

```diff
@@ -933,6 +948,17 @@ async fn l2_update(
             .insert_transaction_data(header.number, &transactions_data, Some(&events_data))
             .context("Insert transaction data into database")?;
 
+        // Associate L1 handler transactions with L2 transactions
+        for tx in &block.transactions {
+            if let TransactionVariant::L1Handler(l1_handler_tx) = &tx.variant {
+                if let Some(l1_tx_hash) = transaction.fetch_and_remove_l1_to_l2_message_log(
+                    &l1_handler_tx.calculate_message_hash(),
+                )? {
+                    transaction.insert_l1_handler_tx(&l1_tx_hash, &tx.hash)?;
+                }
+            }
+        }
+
```

Comment:
There are at least two more places (in P2P checkpoint and tracking sync) where we are inserting transaction data into the database. Can't we maybe move this extra processing inside `Transaction::insert_transaction_data()` instead?